app/MMB/src/scenes/tickets/boardingPasses/mobileTicket/WalletPassenger.js

```diff
@@ -39,7 +61,37 @@ class WalletPassenger extends React.Component<PropsWithContext> {
   };
 
   onPressAndroid = () => {
-    console.warn('TODO');
+    if (this.state.isDownloaded) {
+      console.warn('TODO');
+    } else {
+      this.downloadPkpass();
+    }
+  };
+
+  getFolderPath = () => {
+    const databaseId = idx(this.props.data, _ => _.passenger.databaseId) || '';
+    const flightNumber =
+      this.props.flightNumber == null ? '' : this.props.flightNumber;
+    return `pkpasses/${this.props.bookingId}/${flightNumber}/${databaseId}`;
+  };
+
+  downloadPkpass = async () => {
+    try {
+      const url = idx(this.props.data, _ => _.url) || '';
+      const databaseId = idx(this.props.data, _ => _.passenger.databaseId);
+
+      if (this.props.flightNumber != null && databaseId != null) {
+        const folderPath = this.getFolderPath();
```

Comment:
So if the path is invalid because `databaseId` or `flightNumber` in `getFolderPath` is `''`, then you will just warn that it failed, right?

Reply:
Currently, nothing will happen

Reply:
There could be an else with an alert telling some data is missing 🤷‍♂️

Reply:
I added a couple of TODO's here